source/expressions.tex

```diff
@@ -4824,7 +4824,8 @@
 If the \grammarterm{new-expression} creates an object or an array of
 objects of class type, access and ambiguity control are done for the
 allocation function, the deallocation function\iref{class.free}, and
-the constructor\iref{class.ctor}. If the \grammarterm{new-expression}
+the constructor\iref{class.ctor} selected for the initialization (if any).
+If the \grammarterm{new-expression}
 creates an array of objects of class type, the destructor is potentially
```

Comment:
Does "destructor" here need an "if any" as well?

Reply:
No. The "constructor" part needs an "if any" because we occasionally initialize class objects without calling a constructor (cf. aggregates initialized with braced-init-lists), or at least not invoking a constructor at the syntactic level of /new-expression/ (cf. the example in #2484). In contrast, a class type always has a destructor (it might be trivial, though), and we'll call it (guaranteed) for the elements 0..n-1 if initializing the n-th array element throws an exception. (Note we say "potentially invoked" here, so we're just checking the constraints, but are not making any promises right here that we will actually runtime-call the destructor.)